Assets/SO Architecture/Variables/BaseVariable.cs

```diff
@@ -38,6 +38,7 @@ public virtual T MinClampValue
                     return default(T);
                 }
             }
+            internal set { _minClampedValue = value; }

```

Comment:
Would there be any advantage to enforcing the min and max when they are set?

Reply:
Currently these are only used by the unit tests and not runtime functionality, but I don't see any harm in doing so.

Reply:
Looking at these more closely, it is not possible to do so in `BaseVariable` since the value being set is of generic type `T` and there is not a typed way of being able to clamp this in a way we'd expect at this point in code here. I'm generally fine leaving this as-is since these setters are internal and are only being used by unit tests for verifying other public functionality for setting the value.